catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -210,187 +306,596 @@ private void copyArgumentsToContext(MarshallingContext context,
+                Core.METACARD_PERMISSIONS);
+    }
+
+    private void setLocationFromBoundingBox(MetacardImpl metacard, String north, String south,
+            String east, String west) {
+        try {
+            Envelope envelope = new Envelope(Double.parseDouble(east.trim()),
+                    Double.parseDouble(west.trim()),
+                    Double.parseDouble(south.trim()),
+                    Double.parseDouble(north.trim()));
+            String wkt = WKT_WRITER_THREAD_LOCAL.get()
+                    .write(factory.toGeometry(envelope));
+            if (wkt != null) {
+                metacard.setAttribute(Core.LOCATION, wkt);
             }
+        } catch (NumberFormatException nfe) {
+            LOGGER.debug("Unable to parse double from GMD metadata {}, {}, {}, {}",
+                    west,
+                    east,
+                    south,
+                    north);
         }
+    }
 
+    private void setLocationFromPolygon(String boundingPolygon, MetacardImpl metacard) {
+        try {
+            String[] stringArray = boundingPolygon.split(" ");
+            List<Coordinate> coordinates = new ArrayList<>();
+            for (int i = 0; i < stringArray.length - 1; i += 2) {
+                coordinates.add(new Coordinate(Double.parseDouble(stringArray[i]),
+                        Double.parseDouble(stringArray[i + 1])));
+            }
+
+            LinearRing linearRing =
+                    factory.createLinearRing(coordinates.toArray(new Coordinate[coordinates.size()]));
+            String wkt = WKT_WRITER_THREAD_LOCAL.get()
+                    .write(factory.createPolygon(linearRing, null));
+            if (wkt != null) {
+                metacard.setAttribute(Core.LOCATION, wkt);
+            }
+        } catch (NumberFormatException nfe) {
+            LOGGER.debug("Unable to parse double in {}. Metacard location will not be set.",
+                    boundingPolygon);
+        }
     }
 
-    private void addMetacardSubject(final XstreamPathValueTracker pathValueTracker,
-            MetacardImpl metacard) {
+    private void setLocationFromLineString(String lineString, MetacardImpl metacard) {
+        try {
+            String[] stringArray = lineString.split(" ");
+            List<Coordinate> coordinates = new ArrayList<>();
+            for (int i = 0; i < stringArray.length - 1; i += 2) {
+                coordinates.add(new Coordinate(Double.parseDouble(stringArray[i]),
+                        Double.parseDouble(stringArray[i + 1])));
+            }
+            String wkt = WKT_WRITER_THREAD_LOCAL.get()
+                    .write(factory.createLineString(coordinates.toArray(new Coordinate[coordinates.size()])));
+            if (wkt != null) {
+                metacard.setAttribute(Core.LOCATION, wkt);
+            }
+        } catch (NumberFormatException nfe) {
+            LOGGER.debug("Unable to parse double in {}. Metacard location will not be set.",
```

Comment:
I'd rather not see this sort of exception logged at warn. This could easily fill up the logs if we connect to a source with bad geometry data.  The result here is that location wouldn't be set on the metacard, but the transform still succeeds.  I'd prefer to leave this at debug.